File: 251.cpp
class Vector2D {
public:
    Vector2D(vector<vector<int>>& v) :outer(0), inner(0), data(v) {
        
    }
    
    int next() {
        advanceNext();
        int answer = data[outer][inner];
        inner += 1;
        if (inner == data[outer].size()) {
            outer += 1;
            inner = 0;
        }
        return answer;
    }
    
    bool hasNext() {
        advanceNext();
        return (outer < data.size());
    }
private:
    vector<vector<int>>& data;
    int outer;
    int inner;
    void advanceNext() {
        while (outer < data.size() && data[outer].empty()) {
            outer ++;
            inner = 0;
        }
    }
};

/**
 * Your Vector2D object will be instantiated and called as such:
 * Vector2D* obj = new Vector2D(v);
 * int param_1 = obj->next();
 * bool param_2 = obj->hasNext();
 */

File: 251.py
class Vector2D:

    def __init__(self, v: List[List[int]]):
        self.index = -1
        self.data = []
        self.size = 0
        if not v:
            return
        for lst in v:
            for i in lst:
                self.data.append(i)
        self.size = len(self.data)
        if self.size > 0:
            self.index = 0

    def next(self) -> int:
        if self.hasNext():
            ans = self.data[self.index]
            self.index += 1
            return ans
        else:
            raise(StopIteration)

    def hasNext(self) -> bool:
        if self.index != -1 and self.index < self.size:
            return True
        return False


# Your Vector2D object will be instantiated and called as such:
# obj = Vector2D(v)
# param_1 = obj.next()
# param_2 = obj.hasNext()`
